src/Snap/Internal/Http/Server.hs

```diff
@@ -203,36 +214,58 @@ httpServe defaultTimeout ports localHostname alog' elog' initial handler =
         nports <- mapM bindPort ports
         let socks = map (\x -> case x of ListenHttp s -> s; ListenHttps s _ -> s) nports
 
-        (simpleEventLoop defaultTimeout nports numCapabilities (logE elog) (initial socks)
-                    $ runHTTP defaultTimeout alog elog handler localHostname)
+        (simpleEventLoop defaultTimeout nports numCapabilities (logE errorHandle elog) (initial socks)
+                    $ runHTTP defaultTimeout alog alh elog elh handler localHostname)
           `finally` do
-            logE elog "Server.httpServe: SHUTDOWN"
+            logE errorHandle elog "Server.httpServe: SHUTDOWN"
 
             if initHttps
                 then TLS.stopTLS
                 else return ()
 
-            logE elog "Server.httpServe: BACKEND STOPPED"
+            logE errorHandle elog "Server.httpServe: BACKEND STOPPED"
 
     --------------------------------------------------------------------------
     bindPort (HttpPort  baddr port         ) = bindHttp  baddr port
     bindPort (HttpsPort baddr port cert chainCert key) =
         TLS.bindHttps baddr port cert chainCert key
 
+    errorHandle = fromMaybe defaultErrorLogHandler elh
+
 
 ------------------------------------------------------------------------------
 debugE :: (MonadIO m) => ByteString -> m ()
 debugE s = debug $ "Server: " ++ (map w2c $ S.unpack s)
 
+------------------------------------------------------------------------------
+defaultAccessLogHandler :: AccessLogHandler
```

Comment:
I don't understand, if you simply don't set anything in the config instance then won't this be the default. Do we need to export it if it is the result of the customer doing nothing? I thought that was the beauty of Config being an instance of Monoid.

That said I just wanted to be really sure that was what you wanted me to do and add it to the public api. I'm more than happy to do it if you say yes again. :smile: